main.go

```diff
@@ -83,15 +85,30 @@ func downloadVirtualMachineDiskImage(vmName, volumeName string) error {
 func decompressVirtualMachineDiskImage() error {
 	log.Println("Decompressing downloaded disk image...")
 
-	cmd := exec.Command("gunzip", diskPath)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+	diskImg, err := os.Open(diskPath)
+	if err != nil {
+		return err
+	}
+	defer diskImg.Close()
 
-	if err := cmd.Run(); err != nil {
+	gzipReader, err := gzip.NewReader(diskImg)
+	if err != nil {
+		return err
+	}
+	defer gzipReader.Close()
+
+	newDiskImg, err := os.Create(diskPathDecompressed)
```

Comment:
Does this new image has correct owner and permissions?

Reply:
I've adjusted it to have `0666` that grants read and write permission to everyone.